crates/bevy_hierarchy/src/propagation.rs

```diff
@@ -0,0 +1,109 @@
+use bevy_ecs::{
+    entity::Entity,
+    prelude::{Changed, Component, With, Without},
+    system::Query,
+};
+
+use crate::{Children, Parent};
+
+/// Describe how the global state can be computed and updated from the local state
+pub trait ComputedGlobal {
+    /// Local component associated to this global component
+    type Local: Component;
```

Comment:
This should be `WorldQuery` instead

Reply:
using a `WorldQuery` would increase complexity a lot without bringing a big benefit I think